src/main/java/seedu/address/logic/commands/SearchPublicAddressCommand.java

```diff
@@ -65,10 +70,15 @@ public CommandResult execute(Model model) throws CommandException {
             .toList();
 
 
-        if (personsWithPublicAddressMatch.isEmpty()) {
-            throw new CommandException("No person with the public address found.");
+        if (publicAddressString.length() > 100) { //length of public address too long
+            throw new CommandException(MESSAGE_SEARCH_PUBLIC_ADDRESS_FAILURE_TOO_LONG);
+        } else if (!(publicAddressString.matches("^[a-zA-Z0-9]*$"))) {
+            throw new CommandException(MESSAGE_SEARCH_PUBLIC_ADDRESS_FAILURE_INVALID_CHAR);
+        } else if (publicAddressString.isEmpty()) {
+            throw new CommandException(MESSAGE_SEARCH_PUBLIC_ADDRESS_SUCCESS_NOT_FOUND);
```

Comment:
Can u add this logic into isValidPublicAddress under models/addresses, since i think this checks should be universally used

Reply:
yes sir, that's now added